tests/main/views/test_reset_password.py

```diff
@@ -251,3 +253,191 @@ def test_should_be_an_error_if_send_email_fails(self, send_email):
+
+    def setup_method(self, method):
+        super().setup_method(method)
+
+        data_api_client_config = {'get_user.return_value': self.user(
+            123, "email@email.com", 1234, 'name', 'Name'
+        )}
+
+        self._user = {
+            "user": 123,
+            "email": 'email@email.com',
+        }
+
+        self.data_api_client_patch = mock.patch(
+            'app.main.views.reset_password.data_api_client', **data_api_client_config
+        )
+        self.data_api_client = self.data_api_client_patch.start()
+
+    def teardown_method(self, method):
+        self.data_api_client_patch.stop()
+        super().teardown_method(method)
+
+    @pytest.mark.parametrize(
+        'user_role, redirect_url',
+        [('buyer', '/buyers'), ('supplier', '/suppliers'), ('admin', '/admin')]
+    )
+    def test_change_password_page_displays_correctly(self, user_role, redirect_url):
+        if user_role == 'buyer':
+            self.login_as_buyer()
+        elif user_role.startswith('admin'):
+            self.login_as_admin()
+        else:
+            self.login_as_supplier()
+        response = self.client.get('/user/change-password')
+
+        assert response.status_code == 200
+        document = html.fromstring(response.get_data(as_text=True))
+
+        self.assert_breadcrumbs(response, [("Your account", redirect_url)])
+
+        form_labels = document.xpath('//form//label/text()')
+
+        for label in ['Old password', 'New password', 'Confirm new password']:
+            assert label in form_labels
+
+        assert len(document.xpath('//a[text()="Return to your account"]')) == 1
+
+    @pytest.mark.parametrize(
+        'user_role, redirect_url, user_email',
+        [
+            ('buyer', '/buyers', 'buyer@email.com'),
+            ('supplier', '/suppliers', 'email@email.com'),
+            ('admin', '/admin', 'admin@email.com')
+        ]
+    )
+    @mock.patch('app.main.views.reset_password.DMNotifyClient.send_email')
+    def test_user_can_change_password(self, send_email, user_role, redirect_url, user_email):
+        if user_role == 'buyer':
+            self.login_as_buyer()
+        elif user_role.startswith('admin'):
+            self.login_as_admin()
+        else:
+            self.login_as_supplier()
+        response = self.client.post(
+            '/user/change-password',
+            data={
+                'old_password': '1234567890',
+                'password': '0987654321',
+                'confirm_password': '0987654321'
+            }
+        )
+        assert response.status_code == 302
+        assert response.location == 'http://localhost{}'.format(redirect_url)
+
+        self.data_api_client.update_user_password.assert_called_once_with(123, '0987654321', updater=user_email)
+        self.assert_flashes("You have successfully changed your password.")
+
+        send_email.assert_called_once_with(
+            user_email,
+            template_id='1c4c0562-44aa-4ae4-ba61-e17c544df535',
```

Comment:
Could this id be imported from the config in case it changes?

Reply:
Yep - I'll do the same for the password reset while I'm at it.